gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/shared_policy_group/use_case/UpdateSharedPolicyGroupUseCase.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.shared_policy_group.use_case;
+
+import io.gravitee.apim.core.DomainService;
+import io.gravitee.apim.core.audit.domain_service.AuditDomainService;
+import io.gravitee.apim.core.audit.model.AuditInfo;
+import io.gravitee.apim.core.audit.model.AuditProperties;
+import io.gravitee.apim.core.audit.model.EnvironmentAuditLogEntity;
+import io.gravitee.apim.core.policy.domain_service.PolicyValidationDomainService;
+import io.gravitee.apim.core.shared_policy_group.crud_service.SharedPolicyGroupCrudService;
+import io.gravitee.apim.core.shared_policy_group.exception.SharedPolicyGroupDuplicateCrossIdException;
+import io.gravitee.apim.core.shared_policy_group.model.CreateSharedPolicyGroup;
+import io.gravitee.apim.core.shared_policy_group.model.SharedPolicyGroup;
+import io.gravitee.apim.core.shared_policy_group.model.SharedPolicyGroupAuditEvent;
+import io.gravitee.apim.core.shared_policy_group.model.UpdateSharedPolicyGroup;
+import io.gravitee.common.utils.TimeProvider;
+import io.gravitee.definition.model.v4.flow.step.Step;
+import io.gravitee.rest.api.service.common.UuidString;
+import io.gravitee.rest.api.service.exceptions.InvalidDataException;
+import java.util.Map;
+import lombok.Builder;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@DomainService
+public class UpdateSharedPolicyGroupUseCase {
+
+    private final SharedPolicyGroupCrudService sharedPolicyGroupCrudService;
+    private final PolicyValidationDomainService policyValidationDomainService;
+    private final AuditDomainService auditService;
+
+    public Output execute(Input input) {
+        var existingSharedPolicyGroup =
+            this.sharedPolicyGroupCrudService.getByEnvironmentId(input.auditInfo().environmentId(), input.sharedPolicyGroupId());
+
+        var sharedPolicyGroupToUpdate = SharedPolicyGroup
+            .from(existingSharedPolicyGroup, input.sharedPolicyGroupToUpdate())
```

Comment:
same remark about the naming, if we do not choose `updatedSharedPolicyGroup` maybe an alternative would be `ForUpdate` instead of `ToUpdate`. wdyt ?

Reply:
"to update" I'll name it that it until it's saved in the repository. for or to update same for me. I'm more used to using ToUpdate 🤷‍♂️

Reply:
well i see that we use a bit of both in the code. i leave it like that no problem to change. in my next PRs :)